### PR TITLE
ROT-3440 Fix thread leak

### PR DESCRIPTION
**Pull request by ali-vesalian-deliveryhero**
Reference: https://jira.deliveryhero.com/browse/ROT-3440

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>com.amazonaws</groupId>
   <artifactId>amazon-sqs-java-messaging-lib</artifactId>
-  <version>1.0.8</version>
+  <version>1.0.8-DH-RPS</version>
```
Comment (volkanto) [inline pom.xml]: rps? :D

Comment (vghero) [reply pom.xml]: I was also thinking whether to use `<classifier>` for the patch suffix :thinking:. Maybe also `dh-patch-1` or so :D?

Comment (ali-vesalian-deliveryhero) [reply pom.xml]: Done

File: src/main/java/com/amazon/sqs/javamessaging/SQSSessionCallbackScheduler.java [hunk 2]
```diff
@@ -92,7 +92,7 @@ public void run() {
         try {            
             while (true) {
                 try {
-                    if (closed) {
+                    if (closed || !session.isRunning()) {
```
Comment (volkanto) [inline src/main/java/com/amazon/sqs/javamessaging/SQSSessionCallbackScheduler.java]: btw do we need to adapt the tests for this case?

Comment (ali-vesalian-deliveryhero) [reply src/main/java/com/amazon/sqs/javamessaging/SQSSessionCallbackScheduler.java:95]: Given that this is not our code and we are not fully knowledgable about the code, adapting the tests will be also blind change. The reference of these changes are [this one](https://github.com/awslabs/amazon-sqs-java-messaging-lib/issues/105) and [this](https://github.com/awslabs/amazon-sqs-java-messaging-lib/issues/47) issues. But for testing this change we will run our load test and check the result there

Comment (ali-vesalian-deliveryhero) [reply src/main/java/com/amazon/sqs/javamessaging/SQSSessionCallbackScheduler.java]: I rechecked and changed the PR. Tests now are passed.

**Review by volkanto — APPROVED** [review]
🤞

**Pull request merged**